### PR TITLE
feat(checkout): cart checkout

### PR DESCRIPTION
**Pull request by Davy-Dray**
Cart checkout and get cart total cost

### CODE REVIEW

File: src/cart/cart.module.ts [hunk 1]
```diff
@@ -3,11 +3,15 @@ import { Module } from '@nestjs/common';
 import { Cart, CartSchema } from './schema/cart.schema';
 import { CartService } from './service/cart.service';
 import { CartController } from './controller/cart.controller';
+import { Product, ProductSchema } from '~/products/schema/product.schema';
+import { Order, OrderSchema } from '~/order/schema/order.schema';
```
Comment (iamveekthorr) [inline src/cart/cart.module.ts:8]: I think we can reorder the imports... so we can have other modules first then, we can have other files come in after

File: src/cart/controller/cart.controller.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { RolesGuard } from '~/auth/guards/role.guard';
 import { AddToCartDto } from '../dto/addToCart.dto';
 import { CartService } from '../service/cart.service';
 import { UpdateCartItemQuantityDTO } from '../dto/updateProductQuantity.dto';
+import { CartCheckOutDTO } from '../dto/checkout.dto';
 
 @Controller('carts')
 @Roles(Role.USER)
```
Comment (iamveekthorr) [inline src/cart/controller/cart.controller.ts:16]: You forgot to add the `@UseGuards()`

File: src/cart/service/cart.service.ts [hunk 3]
```diff
@@ -41,14 +50,15 @@ export class CartService {
 
         const productObjectId = new Types.ObjectId(productId);
```
Comment (iamveekthorr) [inline src/cart/service/cart.service.ts:51]: I can't see where you used this variable

File: src/cart/service/cart.service.ts [hunk 4]
```diff
@@ -111,18 +122,104 @@ export class CartService {
...
+
+        for (const item of cart.items) {
+            const product = await this.productModel.findById(item.product);
+
+            if (product) {
+                totalPrice += item.quantity * product.price;
+            } else {
+                throw new AppError(
+                    `Product not found for item with ID: ${item.product}`,
+                    HttpStatus.INTERNAL_SERVER_ERROR
```
Comment (iamveekthorr) [inline src/cart/service/cart.service.ts:215]: This should be a 404 not 500

**Review by iamveekthorr — CHANGES_REQUESTED** [review]
Nice work overall!

File: src/cart/service/cart.service.ts [hunk 5]
```diff
@@ -111,18 +122,104 @@ export class CartService {
...
+            });
+
+            await order.save();
+            await this.cartModel.findByIdAndDelete(cartId);
+
+            return { message: 'Cart checkout successful', orderId: order._id };
+
+        } catch (error) {
+            await session.abortTransaction();
+            throw error;
```
[the remark this answers is not in the document]
Comment (Davy-Dray) [reply src/cart/service/cart.service.ts:178]: I haven't , still researching

File: src/cart/service/cart.service.ts (hunk 3 above)
Comment (Davy-Dray) [reply src/cart/service/cart.service.ts:51]: `    if (existingItem) {
                existingItem.quantity += quantity;
            } else {
                cart.items.push({ product: productObjectId, quantity });
            }`

Comment (Davy-Dray) [reply src/cart/service/cart.service.ts:51]: line 69

**Review by iamveekthorr — APPROVED** [review]

**Pull request merged**